comp/netflow/server/integration_test.go

```diff
@@ -89,12 +91,14 @@ func TestNetFlow_IntegrationTest_NetFlow9(t *testing.T) {
 	epForwarder.EXPECT().SendEventPlatformEventBlocking(gomock.Any(), epforwarder.EventTypeNetworkDevicesNetFlow).Return(nil).Times(29)
 	epForwarder.EXPECT().SendEventPlatformEventBlocking(gomock.Any(), "network-devices-metadata").Return(nil).Times(1)
 
+	time.Sleep(100 * time.Millisecond) // wait to make sure goflow listener is started before sending
```

Comment:
is it possible to do something like:

```
assert.Eventually(func() bool {
   conn = net.Dial(<address of goflow listener>)
})
```

Instead of adding a time.Sleep ?  The problem with this approach is that in the best case it adds 100 milliseconds to each netflow test, which can add up.  In the worst case, if the CI host is very contended, if might take more than 100 milliseconds for the test server to start.

Reply:
@leeavital Indeed `assert.Eventually` might be able to help here. I merge this PR to reduce impact for other teams, but will file another PR to use `assert.Eventually`.

Reply:
Hmmm. goflow listens for UDP packets and doesn't respond to them, and I'm not sure there's a way to tell the difference between a port where nothing is listening to UDP and a port where something is listening but not responding from the outside. We *could* try to start our own listener on the given port and see if it's already in use:
```go
	assert.Eventually(t, func() bool {
		addr := net.JoinHostPort("127.0.0.1",fmt.Sprint(port))
		conn, err := net.ListenPacket("udp", addr)
		if err != nil && strings.Contains(err.Error(), "address already in use") {
			return true
		}
		if conn != nil {
			require.NoError(t, conn.Close())
		}
		return false
	}, time.Second*1, time.Millisecond*10)
```
but then we have another race condition in that if the real listener tries to connect between the ListenPacket call and the Close(), it'll get its own 'address already in use' error.

If we want to support this, I think we'd need to either modify goflow2 to add some sort of ping protocol so that we could send it a message and get a response proving it's still alive, or add some explicit channels to our aggregator and goflow wrappers that they signal when they're ready to receive data.

Reply:
I tried another way here https://github.com/DataDog/datadog-agent/pull/19943 with:

```
	assert.EventuallyWithT(t, func(c *assert.CollectT) {
		// Send netflowV5Data twice to test aggregator
		// Flows will have 2x bytes/packets after aggregation
		packetData, err := testutil.GetNetFlow5Packet()
		require.NoError(c, err, "error getting packet")
		err = testutil.SendUDPPacket(port, packetData)
		require.NoError(c, err, "error sending udp packet")

		netflowEvents, err := flowaggregator.WaitForFlowsToBeFlushed(srv.FlowAgg, 3*time.Second, 2)
		assert.Equal(c, uint64(2), netflowEvents)
		assert.NoError(c, err)
	}, 15*time.Second, 10*time.Millisecond)
```

it doesn't work for the test `TestNetFlow_IntegrationTest_NetFlow5` due to UDP packets sent multiple times with `testutil.SendUDPPacket()`. In this test, we test the payload itself, and since UDP packets are sent multiple times, the number of aggregated NetFlow bytes and packets are not predictable anymore.

Reply:
A fix proposal here:

[netflow] Use Eventually in tests to avoid time.Sleep
https://github.com/DataDog/datadog-agent/pull/19943